src/m365/aad/commands/administrativeunit/administrativeunit-add.ts

```diff
@@ -0,0 +1,82 @@
+import { AdministrativeUnit } from "@microsoft/microsoft-graph-types";
+import GlobalOptions from "../../../../GlobalOptions.js";
+import { Logger } from "../../../../cli/Logger.js";
+import request, { CliRequestOptions } from "../../../../request.js";
+import GraphCommand from "../../../base/GraphCommand.js";
+import commands from "../../commands.js";
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  displayName: string;
+  description?: string;
+  hiddenMembership: boolean;
+}
+
+class AadAdministrativeUnitAddCommand extends GraphCommand {
+  public get name(): string {
+    return commands.ADMINISTRATIVEUNIT_ADD;
+  }
+  public get description(): string {
+    return 'Creates an administrative unit';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        visibility: args.options.visibility
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      {
+        option: '-n, --displayName <displayName>'
+      },
+      {
+        option: '-d, --description [description]'
+      },
+      {
+        option: '--hiddenMembership [hiddenMembership]'
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let administrativeUnit: AdministrativeUnit;
```

Comment:
As we're not using this assigned variable before we created it, let's assign it at creation time: `const administrativeUnit = request.post<etc>`

Reply:
fixed